File: Phonebook.Frontend/src/app/shared/models/classes/Location.ts
import { City } from './City';
import { Room } from './Room';
/* tslint:disable:variable-name */

export class Location {
  public ContactPerson?: any;
  public LinkRoutingWebsite?: any;
  public ReceptionFax?: any;
  public Description?: any;
  public ReceptionPhone?: any;
  public LinkPicture?: any;
  public LinkRoutingInfo?: any;
  public City: City;
  public RoomCollection: Room[];

  constructor(
    city: City,
    roomCollection: Room[],
    contactPerson: any = null,
    linkRoutingWebsite: any = null,
    receptionFax: any = null,
    description: any = null,
    receptionPhone: any = null,
    linkPicture: any = null,
    linkRoutingInfo: any = null
  ) {
    this.City = city;
    this.RoomCollection = roomCollection;
    this.ContactPerson = contactPerson;
    this.LinkRoutingWebsite = linkRoutingWebsite;
    this.ReceptionFax = receptionFax;
    this.Description = description;
    this.ReceptionPhone = receptionPhone;
    this.LinkPicture = linkPicture;
    this.LinkRoutingInfo = linkRoutingInfo;
  }
}

File: Phonebook.Source.PeopleSoft/src/Phonebook.Source.PeopleSoft.Models/Old/Location.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Phonebook.Source.PeopleSoft.Models.Old {
  public class Location {
    private readonly Models.Person person;

    public Location(Models.Person person) {
      this.person = person;
      this.BuildOldLocation();
    }

    private void BuildOldLocation() {
      City = new City(person.Room);
      RoomCollection = new List<Room>(){new Room(person.Room)};
    }

    public dynamic ContactPerson {
      get { return null; }
    }
    public dynamic LinkRoutingWebsite {
      get { return null; }
    }
    public dynamic ReceptionFax {
      get { return null; }
    }
    public dynamic Description {
      get { return null; }
    }
    public dynamic ReceptionPhone {
      get { return null; }
    }
    public dynamic LinkPicture {
      get { return null; }
    }
    public dynamic LinkRoutingInfo {
      get { return null; }
    }
    public City City {
      get;
      private set;
    }
    public IEnumerable<Room>RoomCollection {
      get;
      private set;
    }
  }
}
